### PR TITLE
dev-python/python-gnupg: bump to version 0.4.3

### PR DESCRIPTION
**Pull request by ConiKost**
dev-python/python-gnupg: bump to version 0.4.3

Also bumped to EAPI="7".
Added python 3.7 support.

Closes: https://bugs.gentoo.org/662750
Package-Manager: Portage-2.3.43, Repoman-2.3.10

### CODE REVIEW

File: dev-python/python-gnupg/python-gnupg-0.4.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6,3_7} pypy )
+
+inherit distutils-r1
+
+DESCRIPTION="Python wrapper for GNU Privacy Guard"
+HOMEPAGE="https://pythonhosted.org/python-gnupg/ https://github.com/vsajip/python-gnupg/"
```
Comment (mgorny) [inline dev-python/python-gnupg/python-gnupg-0.4.3.ebuild]: The GitHub thing seems to be a mirror of bitbucket repo.

File: dev-python/python-gnupg/python-gnupg-0.4.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+KEYWORDS="~amd64 ~arm ~ppc ~ppc64 ~x86"
+
+RDEPEND="app-crypt/gnupg"
+DEPEND="${RDEPEND}"
+
+# Tests are still hanging.
+RESTRICT="test"
+
+python_test() {
+	# Note; 1 test fails under pypy only
```
Comment (mgorny) [inline dev-python/python-gnupg/python-gnupg-0.4.3.ebuild]: Is this still the case? If so, please remove pypy from PYTHON_COMPAT.

File: dev-python/python-gnupg/python-gnupg-0.4.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~ppc ~ppc64 ~x86"
+
+RDEPEND="app-crypt/gnupg"
+DEPEND="${RDEPEND}"
+
+# Tests are still hanging.
```
Comment (mgorny) [inline dev-python/python-gnupg/python-gnupg-0.4.3.ebuild]: Did you try to figure out why?

File: dev-python/python-gnupg/python-gnupg-0.4.3.ebuild (hunk 1 above)
Comment (ConiKost) [reply dev-python/python-gnupg/python-gnupg-0.4.3.ebuild]: FYI: I only did a small version bump. So I can't say, why in the past all your mentionted changes have been done. I also never have maintained this package. Anyway, I've removed the URL. Should I do also for `metadata.xml`?

File: dev-python/python-gnupg/python-gnupg-0.4.3.ebuild (hunk 2 above)
Comment (ConiKost) [reply dev-python/python-gnupg/python-gnupg-0.4.3.ebuild]: The don't hang for me, but instead, fail: https://pastebin.com/F3nKVbt9.
Sorry, I am not a python expert, so I don't have a clou..

Comment (ConiKost) [reply dev-python/python-gnupg/python-gnupg-0.4.3.ebuild]: The fail with pypy for me, so removed.
-> https://pastebin.com/8Par3Ke0

File: dev-python/python-gnupg/python-gnupg-0.4.3.ebuild (hunk 3 above)
Comment (ConiKost) [reply dev-python/python-gnupg/python-gnupg-0.4.3.ebuild]: The don't hang for me, but instead, fail: https://pastebin.com/F3nKVbt9.
Sorry, I am not a python expert, so I don't have a clou..

File: dev-python/python-gnupg/python-gnupg-0.4.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6,3_7} )
+
+inherit distutils-r1
+
+DESCRIPTION="Python wrapper for GNU Privacy Guard"
+HOMEPAGE="https://pythonhosted.org/python-gnupg/"
```
[the remark this answers is not in the document]
Comment (ConiKost) [reply dev-python/python-gnupg/python-gnupg-0.4.3.ebuild]: I've added now `https://bitbucket.org/vinay.sajip/python-gnupg/` as the HOMEPAGE. The linked Homepage of pypi seems only to be some sort of a generated docs?

**Comment by ConiKost** [comment]
> Some of the tests definitely fail due to relying on Internet connection. You can remove those.

There is a patch from Debian, I've included it by now. As you said, this won't fix the rests of the test. I would tend to keep the restrict to tests.. Sorry, I don't have the knowledge to fix those.

> However, some of the failures look like legitimate bugs. Looking at upstream bug tracker, this package doesn't work with GPG 2.1.

Are you sure? I see the bug you mean, but on the other side, it seems, according to the changelog, there is some sort of support for 2.1.x? ` Added expect_passphrase password for use on GnuPG >= 2.1 when passing passphrase to gpg via pinentry.`

> Might be easier to look into treecleaning it.

Hm, I wouldn't be happy about it, since my maintained package `sys-apps/onerng` relies on this package and works with the current and this release just fine..

**Comment by mgorny** [comment]
Curious enough, all tests pass when running them locally. Please investigate. `test_gnupg.log` should help.

**Comment by mgorny** [comment]
Figured it out for you. You need to remove the `keys` directory after each run.

**Comment by mgorny** [comment]
Ok, that doesn't solve all the problems. I've made a pull request that fixes race conditions: https://bitbucket.org/vinay.sajip/python-gnupg/pull-requests/24/use-private-temporary-gpg-homes-for-tests/diff

**Comment by mgorny** [comment]
Do you have any custom config in `/etc/gnupg`?

**Comment by ConiKost** [comment]
> Do you have any custom config in /etc/gnupg?

I don't even have such a file. It seems to be a local problem of my desktop, as I did now run the tests on my ThinkPad with Gentoo, and it do pass all tests just fine, as you wrote. So you patch seems to make it work now :)

Tests with `pypy` and `pypy3` are also successful, so (re)added both.

```
>>> Test phase: dev-python/python-gnupg-0.4.3
 * pypy: running distutils-r1_run_phase python_test
.....................s........
----------------------------------------------------------------------
Ran 30 tests in 58.453s

OK (skipped=1)
 * pypy3: running distutils-r1_run_phase python_test
.....................s........
----------------------------------------------------------------------
Ran 30 tests in 60.643s

OK (skipped=1)
 * pypy: running distutils-r1_run_phase _distutils-r1_clean_egg_info
 * pypy3: running distutils-r1_run_phase _distutils-r1_clean_egg_info
>>> Completed testing dev-python/python-gnupg-0.4.3
```

File: dev-python/python-gnupg/python-gnupg-0.4.3.ebuild [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+KEYWORDS="~amd64 ~arm ~ppc ~ppc64 ~x86"
+LICENSE="BSD"
+SLOT="0"
+
+RDEPEND="app-crypt/gnupg"
+DEPEND="${RDEPEND}"
+
+PATCHES=( "${FILESDIR}"/skip_network_needing_test.patch "${FILESDIR}"/use_seperate_keys_directory.patch )
```
[the remark this answers is not in the document]
Comment (ConiKost) [reply dev-python/python-gnupg/python-gnupg-0.4.3.ebuild]: Done.

**Comment by mgorny** [comment]
Aaand thank you!

**Comment by mgorny** [comment]
@ConiKost, please read https://bitbucket.org/vinay.sajip/python-gnupg/pull-requests/24/use-private-temporary-gpg-homes-for-tests#comment-72702264 . I'd appreciate if you could test the upstream patch. I'd also appreciate if you could look if `NO_EXTERNAL_TESTS` is present in our current version and could solve the problem without your patch.

**Comment by ConiKost** [comment]
Ok, will have look and test.

**Comment by ConiKost** [comment]
> I'd appreciate if you could test the upstream patch.

New patch works fine for me. My tests are locally successfull. If I see this correct, per default, the `keys` directory is now dynamic and only static, when `STATIC_TEST_HOMEDIR` is set, which is not the case here.
-> https://bitbucket.org/vinay.sajip/python-gnupg/commits/aeb916c839c0f556dae96bb1136be760ccc19cfe/raw

```
>>> Test phase: dev-python/python-gnupg-0.4.3
 * pypy: running distutils-r1_run_phase python_test
..............................
----------------------------------------------------------------------
Ran 30 tests in 69.823s

OK
 * pypy3: running distutils-r1_run_phase python_test
..............................
----------------------------------------------------------------------
Ran 30 tests in 69.146s

OK
 * python2_7: running distutils-r1_run_phase python_test
..............................
----------------------------------------------------------------------
Ran 30 tests in 66.287s

OK
 * python3_5: running distutils-r1_run_phase python_test
..............................
----------------------------------------------------------------------
Ran 30 tests in 69.256s

OK
 * python3_6: running distutils-r1_run_phase python_test
..............................
----------------------------------------------------------------------
Ran 30 tests in 68.961s

OK
 * python3_7: running distutils-r1_run_phase python_test
..............................
----------------------------------------------------------------------
Ran 30 tests in 63.960s

OK
 * pypy: running distutils-r1_run_phase _distutils-r1_clean_egg_info
 * pypy3: running distutils-r1_run_phase _distutils-r1_clean_egg_info
 * python2_7: running distutils-r1_run_phase _distutils-r1_clean_egg_info
 * python3_5: running distutils-r1_run_phase _distutils-r1_clean_egg_info
 * python3_6: running distutils-r1_run_phase _distutils-r1_clean_egg_info
 * python3_7: running distutils-r1_run_phase _distutils-r1_clean_egg_info
>>> Completed testing dev-python/python-gnupg-0.4.3
```

> I'd also appreciate if you could look if NO_EXTERNAL_TESTS is present in our current version and could solve the problem without your patch.

Unfortunately not, it's not present in current stable 0.4.3. But I've updated the patch, to use the upstream changes and set `NO_EXTERNAL_TESTS` for tests. So with a next new stable version, only the patch has to be dropped.

-> https://bitbucket.org/vinay.sajip/python-gnupg/commits/f16d4e17cf349cf9ee561e117262a4041e4fe2ee/raw

I did now fill a new PR: https://github.com/gentoo/gentoo/pull/9560.
I didn't to a revbump, since I think, a user should't have to reemerge in this case.

**Pull request closed without merging**